app/models/mechanic.rb

```diff
@@ -0,0 +1,10 @@
+class Mechanic < ApplicationRecord
+  has_many :ride_mechanics
+  has_many :rides, through: :ride_mechanics 
+  
+  validates_presence_of :name, :years_experience
+
+  def self.rides_worked_on
```

Comment:
I don't think that you will need a method in here for the rides that a mechanic is working on. You can reach through your `@mechanic` (that you have declared in your `mechanic controller`) to get to rides. So in your view you would call:
```ruby
@mechanic.rides.each do |ride|
```